### ISSUE TITLE
Request for Sigmoid Loss Integration: SigLip

### ISSUE THREAD

**Issue by fabiozappo** [mlfoundations/open_clip#618]
Hi there,

First of all, I'd like to express my appreciation for the outstanding work you've done on this project. Your efforts are truly commendable!

I recently had the opportunity to read the [SigLip](https://arxiv.org/pdf/2303.15343.pdf) research paper, and the authors of the paper propose that replacing the conventional softmax loss with a sigmoid loss could potentially enhance a model's learning capabilities, particularly when dealing with lower batch sizes. Given the potential benefits highlighted in the paper, I'm curious to know if there are any plans to integrate this approach into your repository.

**Comment by rom1504** [comment]
@rwightman Could you describe what you tried on this ?

**Comment by rwightman** [comment]
I had this 80% of the way but hadn't debugged the isend/irecv code to do the neighbour shifting (the paper authors use jax.lax.ppermute for this which is not available as pytorch primitive).

I'll see if I can get it into a state where it does something soonish... and then push a PR for others to look at, help polish/test, etc

**Comment by rwightman** [comment]
@rom1504 @fabiozappo what I have is here, I tried working on the distributed part a bit more, but not sure I've got it behaving properly... at least it seems to not converge, or converge very poorly vs non-distributed ... currently trying a new autograd.Function approach which exchanges grad in opposite dir...
